[rust-server] doesn't support arrays of non-strings in query parameters

<!--
Please follow the issue template below for bug reports and feature requests.
Also please indicate in the issue title which language/library is concerned. Eg:  [JAVA] Bug generating foo with bar 
-->

##### Description

See https://github.com/swagger-api/swagger-codegen/issues/8169#issuecomment-388858171.

It looks as if we're currently assuming that arrays in query parameters must be arrays of strings. I suspect this probably just requires some tinkering with the mustache templates.

```
...
error[E0277]: `std::vec::Vec<i32>` doesn't implement `std::fmt::Display`
     --> src/client/mod.rs:37212:134
      |
37212 |         let query_connections = param_connections.map_or_else(String::new, |query| format!("connections={connections}&", connections=query.join(",")));
      |                                                                                                                                      ^^^^^^^^^^^^^^^ `std::vec::Vec<i32>` cannot be formatted with the default formatter; try using `:?` instead if you are using a format string
      |
      = help: the trait `std::fmt::Display` is not implemented for `std::vec::Vec<i32>`
      = note: required by `std::fmt::Display::fmt`

error[E0277]: the trait bound `std::vec::Vec<i32>: std::str::FromStr` is not satisfied
     --> src/server/mod.rs:37122:71
      |
37122 |                     .filter_map(|param_connections| param_connections.parse::<Vec<i32>>().ok())
      |                                                                       ^^^^^ the trait `std::str::FromStr` is not implemented for `std::vec::Ve
...
```

<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->

##### openapi-generator version

<!-- which version of openapi-generator are you using, is it a regression? -->

##### OpenAPI declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
(here your code)
```
(for YAML code) or
```json
(here your code)
```
(for JSON code), so it becomes more readable. If it is longer than about ten lines,
please create a Gist (https://gist.github.com) or upload it somewhere else and
link it here.
  -->

##### Command line used for generation

<!-- including the language, libraries and various options -->

##### Steps to reproduce

<!-- unambiguous set of steps to reproduce the bug.-->

##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix/enhancement

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

--- a/Cargo.mustache
+++ b/Cargo.mustache
@@ -5,6 +5,7 @@ authors = ["OpenAPI Generator team and contributors"]
 edition = "2018"
 
 [dependencies]
+num-traits = { version = "^0.2", default-features = false }
 serde = "^1.0"
 serde_derive = "^1.0"
 serde_json = "^1.0"

--- a/model.mustache
+++ b/model.mustache
@@ -85,7 +85,7 @@ impl {{{classname}}} {
     pub fn new({{#requiredVars}}{{{name}}}: {{#isNullable}}Option<{{/isNullable}}{{#isEnum}}{{{enumName}}}{{/isEnum}}{{^isEnum}}{{{dataType}}}{{/isEnum}}{{#isNullable}}>{{/isNullable}}{{^-last}}, {{/-last}}{{/requiredVars}}) -> {{{classname}}} {
         {{{classname}}} {
             {{#vars}}
-            {{{name}}}{{^required}}{{#isArray}}: None{{/isArray}}{{#isMap}}: None{{/isMap}}{{^isContainer}}: None{{/isContainer}}{{/required}}{{#required}}{{#isModel}}: Box::new({{{name}}}){{/isModel}}{{/required}},
+                {{{name}}}{{^required}}{{#isArray}}: None{{/isArray}}{{#isMap}}: None{{/isMap}}{{^isContainer}}: None{{/isContainer}}{{/required}}{{#required}}{{#isModel}}: {{#isNullable}}{{{name}}}.map(Box::new){{/isNullable}}{{^isNullable}}Box::new({{{name}}}){{/isNullable}}{{/isModel}}{{/required}},
             {{/vars}}
         }
     }

--- a/reqwest/api.mustache
+++ b/reqwest/api.mustache
@@ -1,9 +1,26 @@
 {{>partial_header}}
 
+use std::fmt::Display;
+
+use num_traits;
 use reqwest;
 
 use crate::apis::ResponseContent;
 use super::{Error, configuration};
+
+
+trait NumVecJoin {
+    fn join(&self, sep: &str) -> String;
+}
+
+impl <T: Display + num_traits::Num> NumVecJoin for Vec<T> {
+    fn join(&self, sep: &str) -> String {
+        self.iter()
+            .map(ToString::to_string)
+            .collect::<Vec<String>>()
+            .join(sep)
+    }
+}
 
 {{#operations}}
 {{#operation}}

